modules/core/04-channel/keeper/upgrade.go

```diff
@@ -944,3 +944,21 @@ func (k Keeper) restoreChannel(ctx sdk.Context, portID, channelID string, upgrad
 
 	return channel
 }
+
+// WriteErrorReceipt will write an error receipt from the provided UpgradeError.
+func (k Keeper) WriteErrorReceipt(ctx sdk.Context, portID, channelID string, upgradeError *types.UpgradeError) {
```

Comment:
Question regarding this new method, is the intension to make `SetUpgradeErrorReceipt` private? seems a little off to have many functions exposed for this. 

I was looking at the cancellation handler and also thought that it is not super clear that an error receipt is set within the `restoreChannel` func above this method. It might be worth while opening another issue to see if we could improve that.

Reply:
yeah I think we can probably make `WriteErrorReceipt` unexported and add a wrapper in `export_test.go`, and potentially just remove the writing of the error receipt from the restore fn, and just call it separately. the error receipt is kind of like a side effect of restoring the channel so I'd be happy with that, will create an issue for it.